### PR TITLE
[ENH] expand surface plotting with new surface object

### LINKED ISSUE #4098: [DOC] experimental-plot-surface-image-and-maskers example does not render well
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed suggestion in detail.

The decoding maps proposed at the end of the example are ugly and not informative. We probably need to tweak viz parameters.
https://nilearn.github.io/stable/auto_examples/08_experimental/plot_surface_image_and_maskers.html#sphx-glr-auto-examples-08-experimental-plot-surface-image-and-maskers-py

### List any pages that would be impacted.

https://nilearn.github.io/stable/auto_examples/08_experimental/plot_surface_image_and_maskers.html

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4098 

Changes proposed in this pull request:

Make sure all surface plotting functions can work with the new surface object

- [x] Adapting surf plotting functions for `SurfaceImage`
- [x] adapt examples/01_plotting/plot_3d_map_to_surface_projection.py
- [x] adapt examples/01_plotting/plot_surf_atlas.py
- [x] adapt examples/01_plotting/plot_surf_stat_map.py
- [x] add functions to return fsaverage data (sulcal, curvature...) as SurfaceImage instance with specific mesh (pial, flat...)
- [x] `bg_map` of the new plotting functions can be a `SurfaceImage` instance

### CODE REVIEW

**Comment by Remi-Gau** [comment]
```
diff --git a/examples/01_plotting/plot_3d_map_to_surface_projection.py b/examples/01_plotting/plot_3d_map_to_surface_projection.py
index 5d709a477..5234d6d56 100644
--- a/examples/01_plotting/plot_3d_map_to_surface_projection.py
+++ b/examples/01_plotting/plot_3d_map_to_surface_projection.py
@@ -57,9 +57,13 @@ texture = surface.vol_to_surf(stat_img, fsaverage.pial_right)
 from nilearn import plotting
 
 fig = plotting.plot_surf_stat_map(
-    fsaverage.infl_right, texture, hemi='right',
-    title='Surface right hemisphere', colorbar=True,
-    threshold=1., bg_map=curv_right_sign,
+    fsaverage.infl_right,
+    texture,
+    hemi="right",
+    title="Surface right hemisphere",
+    colorbar=True,
+    threshold=1.0,
+    bg_map=curv_right_sign,
 )
 fig.show()
 
@@ -72,20 +76,25 @@ fig.show()
 # :func:`~nilearn.plotting.plot_surf_stat_map` to use ``plotly`` instead
 # of ``matplotlib``:
 
-engine = 'plotly'
+engine = "plotly"
 # If plotly is not installed, use matplotlib
 try:
     import plotly.graph_objects as go  # noqa: F401
 except ImportError:
-    engine = 'matplotlib'
+    engine = "matplotlib"
 
 print(f"Using plotting engine {engine}.")
 
 fig = plotting.plot_surf_stat_map(
-    fsaverage.infl_right, texture, hemi='right',
-    title='Surface right hemisphere', colorbar=True,
-    threshold=1., bg_map=curv_right_sign, bg_on_data=True,
-    engine=engine  # Specify the plotting engine here
+    fsaverage.infl_right,
+    texture,
+    hemi="right",
+    title="Surface right hemisphere",
+    colorbar=True,
+    threshold=1.0,
+    bg_map=curv_right_sign,
+    bg_on_data=True,
+    engine=engine,  # Specify the plotting engine here
 )
 
 # Display the figure as with matplotlib figures
@@ -108,26 +117,38 @@ fig = plotting.plot_surf_stat_map(
 # Plot 3D image for comparison
 # ----------------------------
 
-plotting.plot_glass_brain(stat_img, display_mode='r', plot_abs=False,
-                          title='Glass brain', threshold=2.)
+plotting.plot_glass_brain(
+    stat_img,
+    display_mode="r",
+    plot_abs=False,
+    title="Glass brain",
+    threshold=2.0,
+)
 
-plotting.plot_stat_map(stat_img, display_mode='x', threshold=1.,
-                       cut_coords=range(0, 51, 10), title='Slices')
+plotting.plot_stat_map(
+    stat_img,
+    display_mode="x",
+    threshold=1.0,
+    cut_coords=range(0, 51, 10),
+    title="Slices",
+)
 
 # %%
 # Use an atlas and choose regions to outline
 # ------------------------------------------
 
 destrieux_atlas = datasets.fetch_atlas_surf_destrieux()
-parcellation = destrieux_atlas['map_right']
+parcellation = destrieux_atlas["map_right"]
 
 # these are the regions we want to outline
-regions_dict = {b'G_postcentral': 'Postcentral gyrus',
-                b'G_precentral': 'Precentral gyrus'}
+regions_dict = {
+    b"G_postcentral": "Postcentral gyrus",
+    b"G_precentral": "Precentral gyrus",
+}
 
 # get indices in atlas for these labels
 regions_indices = [
-    np.where(np.array(destrieux_atlas['labels']) == region)[0][0]
+    np.where(np.array(destrieux_atlas["labels"]) == region)[0][0]
     for region in regions_dict
 ]
 
@@ -137,16 +158,25 @@ labels = list(regions_dict.values())
 # Display outlines of the regions of interest on top of a statistical map
 # -----------------------------------------------------------------------
 
-figure = plotting.plot_surf_stat_map(fsaverage.infl_right,
-                                     texture, hemi='right',
-                                     title='Surface right hemisphere',
-                                     colorbar=True, threshold=1.,
-                                     bg_map=fsaverage.sulc_right)
+figure = plotting.plot_surf_stat_map(
+    fsaverage.infl_right,
+    texture,
+    hemi="right",
+    title="Surface right hemisphere",
+    colorbar=True,
+    threshold=1.0,
+    bg_map=fsaverage.sulc_right,
+)
 
-plotting.plot_surf_contours(fsaverage.infl_right, parcellation, labels=labels,
-                            levels=regions_indices, figure=figure,
-                            legend=True,
-                            colors=['g', 'k'])
+plotting.plot_surf_contours(
+    fsaverage.infl_right,
+    parcellation,
+    labels=labels,
+    levels=regions_indices,
+    figure=figure,
+    legend=True,
+    colors=["g", "k"],
+)
 plotting.show()
 
 # %%
@@ -158,13 +188,18 @@ plotting.show()
 # the high-resolution fsaverage mesh. Using ``mesh="fsaverage"`` will result
 # in more memory usage and computation time, but finer visualizations.
 
-big_fsaverage = datasets.fetch_surf_fsaverage('fsaverage')
+big_fsaverage = datasets.fetch_surf_fsaverage("fsaverage")
 big_texture = surface.vol_to_surf(stat_img, big_fsaverage.pial_right)
 
-plotting.plot_surf_stat_map(big_fsaverage.infl_right,
-                            big_texture, hemi='right', colorbar=True,
-                            title='Surface right hemisphere: fine mesh',
-                            threshold=1., bg_map=big_fsaverage.sulc_right)
+plotting.plot_surf_stat_map(
+    big_fsaverage.infl_right,
+    big_texture,
+    hemi="right",
+    colorbar=True,
+    title="Surface right hemisphere: fine mesh",
+    threshold=1.0,
+    bg_map=big_fsaverage.sulc_right,
+)
 
 
 # %%
@@ -177,10 +212,12 @@ plotting.plot_surf_stat_map(big_fsaverage.infl_right,
 # :func:`~nilearn.plotting.plot_img_on_surf` projects the images onto
 # `FreeSurfer <https://surfer.nmr.mgh.harvard.edu/>`_\'s fsaverage5.
 
-plotting.plot_img_on_surf(stat_img,
-                          views=['lateral', 'medial'],
-                          hemispheres=['left', 'right'],
-                          colorbar=True)
+plotting.plot_img_on_surf(
+    stat_img,
+    views=["lateral", "medial"],
+    hemispheres=["left", "right"],
+    colorbar=True,
+)
 plotting.show()
 
 # %%
@@ -193,8 +230,9 @@ plotting.show()
 # visualizations in a web browser. See :ref:`interactive-surface-plotting` for
 # more details.
 
-view = plotting.view_surf(fsaverage.infl_right, texture, threshold='90%',
-                          bg_map=fsaverage.sulc_right)
+view = plotting.view_surf(
+    fsaverage.infl_right, texture, threshold="90%", bg_map=fsaverage.sulc_right
+)
 
 # In a Jupyter notebook, if ``view`` is the output of a cell, it will
 # be displayed below the cell
@@ -209,7 +247,7 @@ view
 # We don't need to do the projection ourselves, we can use
 # :func:`~nilearn.plotting.view_img_on_surf`:
 
-view = plotting.view_img_on_surf(stat_img, threshold='90%')
+view = plotting.view_img_on_surf(stat_img, threshold="90%")
 # view.open_in_browser()
 
 view
@@ -231,8 +269,11 @@ destrieux = datasets.fetch_atlas_destrieux_2009(legacy_format=False)
 view = plotting.view_img_on_surf(
     destrieux.maps,
     surf_mesh="fsaverage",
-    vol_to_surf_kwargs={"n_samples": 1, "radius": 0.0,
-                        "interpolation": "nearest"},
+    vol_to_surf_kwargs={
+        "n_samples": 1,
+        "radius": 0.0,
+        "interpolation": "nearest",
+    },
     symmetric_cmap=False,
 )

```

**Comment by Remi-Gau** [comment]
```diff
diff examples/01_plotting/plot_surf_stat_map.py  examples/08_experimental/plot_surf_stat_map_experimental.py

4a5,10
> .. warning::
> 
>     This is an adaption of
>     :ref:`sphx_glr_auto_examples_01_plotting_plot_surf_stat_map.py`
>     to use make it work with the new experimental surface API.
> 
6,7c12,13
< (https://fcon_1000.projects.nitrc.org/indi/enhanced/,
< :footcite:t:`Nooner2012`)
---
> (https://fcon_1000.projects.nitrc.org/indi/enhanced/, :footcite:t:`Nooner2012`
> see :ref:`nki_dataset`)
10,11c16,18
< (https://github.com/fliem/nki_nilearn) and projected onto the Freesurfer
< fsaverage5 template (:footcite:t:`Dale1999` and  :footcite:t:`Fischl1999b`).
---
> (https://github.com/fliem/nki_nilearn)
> and projected onto the Freesurfer fsaverage5 template
> (:footcite:t:`Dale1999` and  :footcite:t:`Fischl1999b`).
18,20c25,27
< Functional connectivity of the seed region to all other cortical nodes in the
< same hemisphere is calculated using Pearson product-moment correlation
< coefficient.
---
> Functional connectivity of the seed region to all other cortical nodes
> in the same hemisphere is calculated
> using Pearson product-moment correlation coefficient.
36c43,48
< from nilearn import datasets
---
> from nilearn.experimental.surface import (
>     fetch_destrieux,
>     fetch_nki,
>     load_fsaverage,
>     load_fsaverage_data,
> )
38c50
< nki_dataset = datasets.fetch_surf_nki_enhanced(n_subjects=1)
---
> nki_dataset = fetch_nki(n_subjects=1)
40,45c52,53
< # The nki dictionary contains file names for the data
< # of all downloaded subjects.
< print(
<     "Resting state data of the first subjects on the "
<     f"fsaverag5 surface left hemisphere is at: {nki_dataset['func_left'][0]}"
< )
---
> # The nki list contains a SurfaceImage instance
> # for the data of each subject along with fsaverage pial meshes.
48,50c56,57
< destrieux_atlas = datasets.fetch_atlas_surf_destrieux()
< parcellation = destrieux_atlas["map_left"]
< labels = destrieux_atlas["labels"]
---
> destrieux_atlas, labels = fetch_destrieux()
> parcellation = destrieux_atlas.data.parts["left"]
53c60
< fsaverage = datasets.fetch_surf_fsaverage()
---
> fsaverage_meshes = load_fsaverage()
55,64c62
< # The fsaverage dataset contains file names pointing to
< # the file locations
< print(
<     "Fsaverage5 pial surface of left hemisphere is at: "
<     f"{fsaverage['pial_left']}"
< )
< print(
<     "Fsaverage5 flatten pial surface of left hemisphere is at: "
<     f"{fsaverage['flat_left']}"
< )
---
> # The fsaverage meshes contains the FileMesh objects:
66,67c64,65
<     "Fsaverage5 inflated surface of left hemisphere is at: "
<     f"{fsaverage['infl_left']}"
---
>     "Fsaverage5 pial surface of left hemisphere is: "
>     f"{fsaverage_meshes['pial'].parts['left']}"
70,71c68,69
<     "Fsaverage5 sulcal depth map of left hemisphere is at: "
<     f"{fsaverage['sulc_left']}"
---
>     "Fsaverage5 inflated surface of left hemisphere is: "
>     f"{fsaverage_meshes['flat'].parts['left']}"
74,75c72,73
<     "Fsaverage5 curvature map of left hemisphere is at: "
<     f"{fsaverage['curv_left']}"
---
>     "Fsaverage5 inflated surface of left hemisphere is: "
>     f"{fsaverage_meshes['inflated'].parts['left']}"
77a76,82
> # The fsaverage data contains SurfaceImage instances with meshes and data
> fsaverage_sulcal = load_fsaverage_data(data_type="sulcal")
> print(f"Fsaverage5 sulcal depth map: {fsaverage_sulcal}")
> 
> fsaverage_curvature = load_fsaverage_data(data_type="curvature")
> print(f"Fsaverage5 sulcal curvature map: {fsaverage_curvature}")
> 
83,85c88
< from nilearn import surface
< 
< timeseries = surface.load_surf_data(nki_dataset["func_left"][0])
---
> timeseries = nki_dataset[0].data.parts["left"].T
119c122,125
< from nilearn import plotting
---
> from nilearn.experimental import plotting
> from nilearn.plotting import show
> 
> print(f"{type(nki_dataset[0].mesh)=}")
122c128
<     fsaverage["pial_left"],
---
>     surf_mesh=nki_dataset[0].mesh.parts["left"],
126c132
<     bg_map=fsaverage["sulc_left"],
---
>     bg_map=fsaverage_sulcal,
130a137,138
> show()
> 
133c141,142
< # of interest on the cortex. To make this plot easier to read,
---
> # of interest on the cortex.
> # To make this plot easier to read,
136c145
< bg_map = np.sign(surface.load_surf_data(fsaverage["curv_left"]))
---
> bg_map = np.sign(fsaverage_curvature.data.parts["left"])
142c151
<     fsaverage["flat_left"],
---
>     surf_mesh=fsaverage_meshes["flat"].parts["left"],
146c155
<     bg_map=bg_map_rescaled,
---
>     bg_map=fsaverage_sulcal,
154c163
<     fsaverage["pial_left"],
---
>     surf_mesh=nki_dataset[0].mesh.parts["left"],
159c168
<     bg_map=fsaverage["sulc_left"],
---
>     bg_map=fsaverage_sulcal,
164a174,175
> show()
> 
169c180
<     fsaverage["pial_left"],
---
>     surf_mesh=nki_dataset[0].mesh.parts["left"],
174c185
<     bg_map=fsaverage["sulc_left"],
---
>     bg_map=fsaverage_sulcal,
180a192,193
> show()
> 
183,186c196,199
< # To capture 3D structure without depth information, the default is to plot a
< # half transparent surface.
< # Note that you can also control the transparency with a background map using
< # the alpha parameter.
---
> # To capture 3D structure without depth information,
> # the default is to plot a half transparent surface.
> # Note that you can also control the transparency
> # with a background map using the alpha parameter.
188c201
<     fsaverage["pial_left"],
---
>     surf_mesh=nki_dataset[0].mesh.parts["left"],
197a211,212
> show()
> 
208c223
<     fsaverage["infl_left"],
---
>     surf_mesh=fsaverage_meshes["inflated"].parts["left"],
211c226
<     bg_map=fsaverage["sulc_left"],
---
>     bg_map=fsaverage_sulcal,
218c233
< plotting.show()
---
> show()

```

**Comment by Remi-Gau** [comment]
```diff
diff examples/01_plotting/plot_3d_map_to_surface_projection.py  examples/08_experimental/plot_3d_map_to_surface_projection_experimental.py 

4a5,10
> .. warning::
> 
>     This is an adaption of
>     :ref:`sphx_glr_auto_examples_01_plotting_plot_3d_map_to_surface_projection.py`
>     to use make it work with the new experimental surface API.
> 
6,8c12,16
< using :func:`~nilearn.surface.vol_to_surf`, display a surface plot of the
< projected map using :func:`~nilearn.plotting.plot_surf_stat_map` with
< different plotting engines, and add contours of regions of interest using
---
> using :func:`~nilearn.surface.vol_to_surf`,
> display a surface plot of the projected map
> using :func:`~nilearn.plotting.plot_surf_stat_map`
> with different plotting engines,
> and add contours of regions of interest using
10d17
< 
26c33,35
< fsaverage = datasets.fetch_surf_fsaverage()
---
> from nilearn.experimental.surface import load_fsaverage, load_fsaverage_data
> 
> fsaverage_meshes = load_fsaverage()
33,34c42,44
< # and define a surface map whose value for a given :term:`vertex` is
< # 1 if the curvature is positive, -1 if the curvature is negative.
---
> # and define a surface map whose value for a given :term:`vertex`
> # is 1 if the curvature is positive,
> # -1 if the curvature is negative.
40,41c50,51
< curv_right = surface.load_surf_data(fsaverage.curv_right)
< curv_right_sign = np.sign(curv_right)
---
> fsaverage_curvature = load_fsaverage_data(data_type="curvature")
> curv_right_sign = np.sign(fsaverage_curvature.data.parts["right"])
45a56
> from nilearn.experimental.surface import SurfaceImage
47c58,68
< texture = surface.vol_to_surf(stat_img, fsaverage.pial_right)
---
> img = SurfaceImage(
>     mesh=fsaverage_meshes["pial"],
>     data={
>         "left": surface.vol_to_surf(
>             stat_img, fsaverage_meshes["pial"].parts["left"]
>         ),
>         "right": surface.vol_to_surf(
>             stat_img, fsaverage_meshes["pial"].parts["right"]
>         ),
>     },
> )
57c78
< from nilearn import plotting
---
> from nilearn.experimental.plotting import plot_surf_stat_map
59,61c80,81
< fig = plotting.plot_surf_stat_map(
<     fsaverage.infl_right,
<     texture,
---
> fig = plot_surf_stat_map(
>     stat_map=img,
63c83
<     title="Surface right hemisphere",
---
>     title="Surface with matplotlib",
69a90
> 
76,77c97,98
< # :func:`~nilearn.plotting.plot_surf_stat_map` to use ``plotly`` instead
< # of ``matplotlib``:
---
> # :func:`~nilearn.plotting.plot_surf_stat_map` to use ``plotly``
> # instead of ``matplotlib``:
88,90c109,110
< fig = plotting.plot_surf_stat_map(
<     fsaverage.infl_right,
<     texture,
---
> fig = plot_surf_stat_map(
>     stat_map=img,
92c112
<     title="Surface right hemisphere",
---
>     title="Surface with plotly",
101c121
< # fig.show()
---
> fig.show()
105,110c125,131
< # :class:`matplotlib.figure.Figure` is returned. With ``plotly`` as the
< # plotting engine, a custom
< # :class:`~nilearn.plotting.displays.PlotlySurfaceFigure` is returned which
< # provides a similar API to the :class:`~matplotlib.figure.Figure`.
< # For example, you can save a static version of the figure to file (this
< # option requires to have ``kaleido`` installed):
---
> # :class:`matplotlib.figure.Figure` is returned.
> # With ``plotly`` as the plotting engine,
> # a custom :class:`~nilearn.plotting.displays.PlotlySurfaceFigure`
> # is returned which provides a similar API
> # to the :class:`~matplotlib.figure.Figure`.
> # For example, you can save a static version of the figure to file
> # (this option requires to have ``kaleido`` installed):
112c133
< # Save the figure as we would do with a matplotlib figure
---
> # Save the figure as we would do with a matplotlib figure.
119a141,142
> from nilearn import plotting
> 
121c144
<     stat_img,
---
>     stat_map_img=stat_img,
129c152
<     stat_img,
---
>     stat_map_img=stat_img,
140,141c163,165
< destrieux_atlas = datasets.fetch_atlas_surf_destrieux()
< parcellation = destrieux_atlas["map_right"]
---
> from nilearn.experimental.surface import fetch_destrieux
> 
> destrieux_atlas, label_names = fetch_destrieux(mesh_type="inflated")
151,152c175
<     np.where(np.array(destrieux_atlas["labels"]) == region)[0][0]
<     for region in regions_dict
---
>     np.where(np.array(label_names) == region)[0][0] for region in regions_dict
161,163c184,190
< figure = plotting.plot_surf_stat_map(
<     fsaverage.infl_right,
<     texture,
---
> from nilearn.experimental.plotting import plot_surf_contours
> 
> fsaverage_sulcal = load_fsaverage_data(data_type="sulcal")
> 
> figure = plot_surf_stat_map(
>     stat_map=img,
>     surf_mesh=fsaverage_meshes["inflated"],
165c192
<     title="Surface right hemisphere",
---
>     title="ROI outlines on surface",
168c195
<     bg_map=fsaverage.sulc_right,
---
>     bg_map=fsaverage_sulcal,
171,173c198,199
< plotting.plot_surf_contours(
<     fsaverage.infl_right,
<     parcellation,
---
> plot_surf_contours(
>     roi_map=destrieux_atlas,
187,188c213,215
< # which specifies whether to fetch the low-resolution ``fsaverage5`` mesh, or
< # the high-resolution fsaverage mesh. Using ``mesh="fsaverage"`` will result
---
> # which specifies whether to fetch the low-resolution ``fsaverage5`` mesh,
> # or the high-resolution fsaverage mesh.
> # Using ``mesh="fsaverage"`` will result
191,192c218,222
< big_fsaverage = datasets.fetch_surf_fsaverage("fsaverage")
< big_texture = surface.vol_to_surf(stat_img, big_fsaverage.pial_right)
---
> big_fsaverage_meshes = load_fsaverage("fsaverage")
> 
> big_fsaverage_sulcal = load_fsaverage_data(
>     mesh_name="fsaverage", data_type="sulcal"
> )
194,196c224,237
< plotting.plot_surf_stat_map(
<     big_fsaverage.infl_right,
<     big_texture,
---
> big_img = SurfaceImage(
>     mesh=big_fsaverage_meshes["pial"],
>     data={
>         "left": surface.vol_to_surf(
>             stat_img, big_fsaverage_meshes["pial"].parts["left"]
>         ),
>         "right": surface.vol_to_surf(
>             stat_img, big_fsaverage_meshes["pial"].parts["right"]
>         ),
>     },
> )
> 
> plot_surf_stat_map(
>     big_img,
199c240
<     title="Surface right hemisphere: fine mesh",
---
>     title="Surface fine mesh",
201c242
<     bg_map=big_fsaverage.sulc_right,
---
>     bg_map=big_fsaverage_sulcal,
209,211c250,254
< # :func:`~nilearn.plotting.plot_img_on_surf` takes a statistical map and
< # projects it onto a surface. It supports multiple choices of orientations,
< # and can plot either one or both hemispheres. If no ``surf_mesh`` is given,
---
> # :func:`~nilearn.plotting.plot_img_on_surf` takes a statistical map
> # and projects it onto a surface.
> # It supports multiple choices of orientations,
> # and can plot either one or both hemispheres.
> # If no ``surf_mesh`` is given,
219a263,265
>     cmap="seismic",
>     title="multiple views of the 3D volume",
>     bg_on_data=True,
229,234c275,287
< # :func:`nilearn.plotting.view_img_on_surf` that give more interactive
< # visualizations in a web browser. See :ref:`interactive-surface-plotting` for
< # more details.
< 
< view = plotting.view_surf(
<     fsaverage.infl_right, texture, threshold="90%", bg_map=fsaverage.sulc_right
---
> # :func:`nilearn.plotting.view_img_on_surf` that give
> # more interactive visualizations in a web browser.
> # See :ref:`interactive-surface-plotting` for more details.
> 
> from nilearn.experimental.plotting import view_surf
> 
> view = view_surf(
>     surf_mesh=fsaverage_meshes["inflated"],
>     surf_map=img,
>     threshold="90%",
>     bg_map=fsaverage_sulcal,
>     hemi="right",
>     title="3D visualization in a web browser",
237,238c290,291
< # In a Jupyter notebook, if ``view`` is the output of a cell, it will
< # be displayed below the cell
---
> # In a Jupyter notebook, if ``view`` is the output of a cell,
> # it will be displayed below the cell
240,246c293
< 
< # %%
< 
< # uncomment this to open the plot in a web browser:
< # view.open_in_browser()
< 
< # %%
---
> datasets

```

**Comment by Remi-Gau** [comment]
Doc build is failing because of the same issue as reported in here: https://github.com/nilearn/nilearn/issues/4379

**Comment by bthirion** [comment]
I'm happy with the changes( I did not spot any pb by reading through the code), but I find it embarassing to merge it without seeing the examples.

**Comment by Remi-Gau** [comment]
> I'm happy with the changes( I did not spot any pb by reading through the code), but I find it embarassing to merge it without seeing the examples.


Yeah I know. I want to get back to fixing our doc CI deploy once I have dealt with some of the surface stuff.

In the meantime:

Each documentation build workflow contains links that allows for download of the built doc.

For example here: 

URL github
https://github.com/nilearn/nilearn/actions/runs/9380986566/job/25829503823#step:15:37
nightly URL
https://github.com/nilearn/nilearn/actions/runs/9380986566/job/25829503823#step:16:1

So you can still download check the examples:
https://nightly.link/nilearn/nilearn/actions/runs/9380986566

@man-shu ☝🏾 in case you need to check some of your doc build!

**Comment by man-shu** [comment]
> @man-shu ☝🏾 in case you need to check some of your doc build!

Yes! Thanks! Good to know.

File: nilearn/experimental/plotting/_surface_plotting.py [hunk 1]
```diff
@@ -1,36 +1,203 @@
...
+    if isinstance(bg_map, SurfaceImage):
+        assert bg_map.data.parts[hemi] is not None
+        bg_map = bg_map.data.parts[hemi]
+    return bg_map
+
+
+def _check_hemi_present(mesh: PolyMesh, hemi: str) -> None:
+    """Check that a given hemisphere exists both in data and mesh."""
+    if hemi not in mesh.parts:
+        raise ValueError(f"{hemi} must be present in mesh")
```
Comment (man-shu) [inline nilearn/experimental/plotting/_surface_plotting.py:57]: I wonder if some of the codecov complaints might actually be making sense this time.
Or is it that we only want to add tests for these once we make the surface stuff public?

**Comment by Remi-Gau** [comment]
TODO:

Some fixes need to be made to match : 
- https://nilearn.github.io/dev/auto_examples/01_plotting/plot_3d_map_to_surface_projection.html

File: examples/08_experimental/plot_surface_image_and_maskers.py [hunk 2]
```diff
@@ -52,8 +53,8 @@
 for view, ax_row in zip(views, axes):
     for ax, hemi in zip(ax_row, hemispheres):
         plotting.plot_surf(
-            mean_img,
-            part=hemi,
+            surf_map=mean_img,
+            hemi=hemi,
             view=view,
             figure=fig,
             axes=ax,
```
Comment (bthirion) [inline examples/08_experimental/plot_surface_image_and_maskers.py:60]: I think that this figure would render better if se use 'bg_on_data=True' ? Alternatively, we could also threshold the image.

**Review by bthirion — COMMENTED** [review]
The documentation renders really well. Beautiful work !
I only have one minor comment left.

**Comment by Remi-Gau** [comment]
To fix: ROIs outline are wrong in this figure.

plot-3d-map-to-surface-projection-experimental-py

image

**Comment by Remi-Gau** [comment]
Also some issues on the higher res version 'fine mesh' compare to original.

How it looks now:

image


How it looks in the original example

https://nilearn.github.io/dev/auto_examples/01_plotting/plot_3d_map_to_surface_projection.html#plot-with-higher-resolution-mesh

image

File: examples/08_experimental/plot_surface_image_and_maskers.py (hunk 2 above)
Comment (Remi-Gau) [reply examples/08_experimental/plot_surface_image_and_maskers.py:60]: done but the new plot does not look tons better

**Comment by bthirion** [comment]
LMK when the issue is fixed. the figures had not been generated in the version of the doc that I found on the CI.

**Comment by Remi-Gau** [comment]
checked the latest build of the doc: https://github.com/nilearn/nilearn/actions/runs/9419801486/job/25950563153?pr=4431#step:15:37

Things look fixed now.

This is ready for final review on my side.

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**